kb/2025-10-07/velero-backup-filesystem-freeze.md

```diff
@@ -0,0 +1,193 @@
+* **Windows guests**: Require the **VSS** service to be enabled for filesystem freeze functionality
+
+> **Important**: Filesystem freeze/thaw functionality depends on guest VM configuration, which is outside Harvester's control. Users are responsible for ensuring compatibility before implementing Velero backup hooks with filesystem freeze.
+
+### Verifying Filesystem Freeze Compatibility
+
+To confirm that your VM supports filesystem freeze operations:
+
+1. Access the virtual machine's virt-launcher `compute` container:
+
+   ```bash
+   POD=$(kubectl get pods -n default \
+     -l vm.kubevirt.io/name=vm1 \
+     -o jsonpath='{.items[0].metadata.name}')
+   kubectl exec -it $POD -n default -c compute -- bash
+   ```
+
+2. Test filesystem freeze using the [virt-freezer](https://github.com/kubevirt/kubevirt/blob/main/docs/freeze.md#virt-freezer) application available in the `compute` container:
+
+   ```bash
+   virt-freezer --freeze --namespace <VM namespace> --name <VM name>
+   ```
+
+3. **Critical**: Always verify the freeze operation result and thaw the VM filesystems before performing any other operations
+
+## Prerequisites
+
+All preparation steps outlined in [External CSI Storage Backup and Restore With Velero](https://harvesterhci.io/kb/2025/05/26/velero-backup-restore) are mandatory, including:
+
+- Harvester installation and configuration
+- Velero installation and setup
+- S3-compatible storage configuration
+- Proper networking and permissions
+
+## Implementing Filesystem Freeze Hooks for VM Backup Consistency
+
+Velero supports pre and post backup hooks that can be integrated with KubeVirt's **virt-freezer** to ensure filesystem consistency during VM backups.
+
+### Configuring VM Template Annotations
+
+For all VMs requiring data consistency, add the following annotations to the VM template:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: vm-nfs
+  namespace: demo
+spec:
+  template:
+    metadata:
+      annotations:
+        # These annotations will be applied to the virt-launcher pod
+        pre.hook.backup.velero.io/command: '["/usr/bin/virt-freezer", "--freeze", "--namespace", "[VM Namespace]", "--name", "[VM Name]"'
+        pre.hook.backup.velero.io/container: compute
+        pre.hook.backup.velero.io/on-error: Fail
+        pre.hook.backup.velero.io/timeout: 30s
+        
+        post.hook.backup.velero.io/command: '["/usr/bin/virt-freezer", "--unfreeze", "--namespace", "[VM Namespace]", "--name", "[VM Name]"]'
+        post.hook.backup.velero.io/container: compute
+        post.hook.backup.velero.io/timeout: 30s
+    spec:
+      # ...rest of VM spec...
+```
+
+These annotations will be propagated to the related virt-launcher pod and instruct Velero to:
+- **Freeze** the VM filesystem before backup creation begins
+- **Thaw** the VM filesystem after backup completion
+
+**Important**: Replace `[VM Namespace]` and `[VM Name]` with the actual namespace and name of your VM.
```

Comment:
Use a consistent way to name your variables in examples, `[VM Namespace]` or either `<VM Namespace>`.